src/main/scala/com/amazon/deequ/analyzers/Histogram.scala

```diff
@@ -125,13 +126,115 @@ case class Histogram(
       case _ => data
     }
   }
+
+  private def query(data: DataFrame): DataFrame = {
+    aggregateFunction.query(this.column, data)
+  }
 }
 
 object Histogram {
   val NullFieldReplacement = "NullValue"
   val MaximumAllowedDetailBins = 1000
+  val count_function = "count"
+  val sum_function = "sum"
+
+
+  def createAggregateFunction(function: String, aggregateColumn: String): AggregateFunction = {
+    function match {
+      case Histogram.count_function => Count
+      case Histogram.sum_function => Sum(aggregateColumn)
+      case _ => throw new IllegalArgumentException("Wrong aggregate function name: " + function)
+    }
+  }
```

Comment:
Can we remove this function and let the caller of the function provide a valid instance of `AggregateFunction` ? The reason I say this is because if we add more subclasses of `AggregateFunction` in the future, we will need to update this `createAggregateFunction` to accept the superset of ALL parameters for all subtypes.

Reply:
This function is used to deserialize object from json. I can move it to AnalysisResultSerde, but there must be a piece of code that deserializes all combinations from json.

Reply:
Sure, I think it is fine then to keep it outside of `Histogram` and in `AnalysisResultSerde`.